protocol/synthetix/contracts/modules/core/RewardsManagerModule.sol

```diff
@@ -235,11 +235,12 @@ contract RewardsManagerModule is IRewardsManagerModule {
             revert ParameterError.InvalidParameter("distributor", "is not registered");
         }
 
-        rewardIds.remove(rewardId);
+        pool.vaults[collateralType].rewardIds.remove(rewardId);
```

Comment:
This has the same effect of the line that was removed, and actually costs a little more gas because it does not use the pointer declared in line 226.

I think what the auditors meant was to identify the RewardDistribution object to which the rewardId in question refers to, and resetting that. Since `_getRewardId` produces an id that is determined by `poolId, collateralType, msg.sender`, a future reward id could use the same id of one that was removed, and thus be initialized with pre-existing data.

My recommendation would be to introduce a `resetEntry` function in RewardDistribution, which would set all of this object's properties to zero.